app/packages/files/provider-file.ts

```diff
@@ -0,0 +1,37 @@
+import { tracked } from '@glimmer/tracking';
+import FileProviderModel from 'ember-osf-web/models/file-provider';
+
+export default abstract class File {
```

Comment:
Should this be exporting `ProviderFile` or `Provider` (since `ProviderFile` sounds a bit misleading) ? Also it seems to share many similarities withe `File` abstract class. Can this implement `File` instead of being a class on its own?

Reply:
Yeah, it should probably be exporting `ProviderFile`. I called it `ProviderFile` instead of `Provider` because we're using it for its file capabilities, rather than as a provider, per se. I could see either way, but I lean towards `ProviderFile` personally.

Since the `ProviderFile` is more like a subset of the `File` functionality, we'd have to have `File` derive from `ProviderFile`, but honestly that just seems a bit weird.

Reply:
In which case, can the `ProviderFile` implement noop for the functionalities that is supported by `File` but not by `ProviderFile`?

Reply:
I tried this a bit in the beginning, and I'm trying again now, and it really complicates things to try to make these related. The problem is that the FileProviderModel doesn't have fields that the FileModel has, and this is going to involve jumping through a lot of messy hoops to make the typescript work out right. Could we just keep these as very-similar-but-not-related interfaces?

Reply:
NVM then. Not worth the effort.